### ISSUE TITLE
NullPointer occur int the method DashManifest.getPeriod(int)

### ISSUE THREAD

**Issue by jiafei1986** [google/ExoPlayer#3268]
*** ISSUES THAT IGNORE THIS TEMPLATE WILL BE CLOSED WITHOUT INVESTIGATION ***

### Issue description
We are using exoplayer to play dash with DRM.  We checked the firebase and found some null pointer from the internal exoplayer

### Reproduction steps
We just get feedback from our customer and checked firebase log, we can find some null pointer errors. 

This error happened when our customer playing the dash drm.  then we checked the firebase log.

### Link to test content
This is a live stream.

### Version of ExoPlayer being used
2.4.1

### Device(s) and version(s) of Android being used
It could be found on any devices, I just list down some but not all device. 
SM-T805 SM-G930F  SM-G935F

### A full bug report captured from the device
Crash report in the firebase:

java.lang.NullPointerException: Attempt to invoke virtual method 'com.google.android.exoplayer2.e.b.a.d com.google.android.exoplayer2.e.b.a.b.a(int)' on a null object reference

Mapping file:
`com.google.android.exoplayer2.source.dash.manifest.DashManifest -> com.google.android.exoplayer2.e.b.a.b:
    long availabilityStartTime -> a
    long duration -> b
    long minBufferTime -> c
    boolean dynamic -> d
    long minUpdatePeriod -> e
    long timeShiftBufferDepth -> f
    long suggestedPresentationDelay -> g
    com.google.android.exoplayer2.source.dash.manifest.UtcTimingElement utcTiming -> h
    android.net.Uri location -> i
    java.util.List periods -> j
    53:64:void <init>(long,long,long,boolean,long,long,long,com.google.android.exoplayer2.source.dash.manifest.UtcTimingElement,android.net.Uri,java.util.List) -> <init>
    67:67:int getPeriodCount() -> a
    71:71:com.google.android.exoplayer2.source.dash.manifest.Period getPeriod(int) -> a
    75:77:long getPeriodDurationMs(int) -> b
    81:81:long getPeriodDurationUs(int) -> c`

It seems the method com.google.android.exoplayer2.source.dash.manifest.Period **getPeriod**(int) get nullpointer. 

Let me know if you need more information.  
Thanks in advance.

**Comment by ojw28** [comment]
Please provide a proper de-obfuscated stack trace showing the failure.

**Comment by jiafei1986** [comment]
@ojw28  Thank you for your reply, we found now only one line string sent to the firebase, we will send the whole stack trace to firebase and post here. As this error only happened on our customer side, we cannot replicate it.  so will post stack trace once get from firebase.

**Comment by jiafei1986** [comment]
@ojw28  here is the stack trace:
```playerFailed [1.85]
                                                                      com.google.android.exoplayer2.ExoPlaybackException
                                                                          at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:362)
                                                                          at android.os.Handler.dispatchMessage(Handler.java:98)
                                                                          at android.os.Looper.loop(Looper.java:145)
                                                                          at android.os.HandlerThread.run(HandlerThread.java:61)
                                                                       Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'com.google.android.exoplayer2.source.dash.manifest.Period com.google.android.exoplayer2.source.dash.manifest.DashManifest.getPeriod(int)' on a null object reference
                                                                          at com.google.android.exoplayer2.source.dash.DashMediaSource.createPeriod(DashMediaSource.java:286)
                                                                          at com.google.android.exoplayer2.ExoPlayerImplInternal$MediaPeriodHolder.<init>(ExoPlayerImplInternal.java:1472)
                                                                          at com.google.android.exoplayer2.ExoPlayerImplInternal.maybeUpdateLoadingPeriod(ExoPlayerImplInternal.java:1289)
                                                                          at com.google.android.exoplayer2.ExoPlayerImplInternal.updatePeriods(ExoPlayerImplInternal.java:1138)
                                                                          at com.google.android.exoplayer2.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:460)
                                                                          at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:308)
                                                                          at android.os.Handler.dispatchMessage(Handler.java:98) 
                                                                          at android.os.Looper.loop(Looper.java:145) 
                                                                          at android.os.HandlerThread.run(HandlerThread.java:61) ```

**Comment by ojw28** [comment]
The only way I can see that happening is if your application code is re-using the same `DashMediaSource` instance more than once. As per the [guide](https://google.github.io/ExoPlayer/guide.html#mediasource), if you want to prepare the player to play a particular piece of media more than once, you need to instantiate a fresh `MediaSource` instance each time. Please take a look at your code and see whether that's the issue.

**Comment by jiafei1986** [comment]
@ojw28  thank you for your reply, I checked the code, there is a logic that our app will retry to play when it gets some error,  we implemented the retry function in the EventListener.onPlayerError(), the logic is that release the player, and prepare the player again.  the code of release and prepare the player same as the demo player. i will check the code and let you know whether we create a new MediaSource for each retry.
